app/jsx/components/Search2Comp.jsx

```diff
@@ -1,27 +1,79 @@
-// ##### Search Component ##### //
+// ##### Search Component - With Search Controls ##### //
 
 import React from 'react'
 import Form from 'react-router-form'
 
+class CustomRadioInput extends React.Component {
+  render() {
+    return (
+      <input type="radio" id="c-search2__refine-campus" name={this.props.name} value={this.props.value} onClick={this.props.handleRadioSelect}/>
+  )}
+}
+
+class SearchControls extends React.Component {
+  constructor(props) {
+    super(props)
+    this.handleRadioSelect = this.props.handleRadioSelect.bind(this)
+  }
+
+  handleRadioSelect(){
+    this.props.handleRadioSelect()
+  }
+
+  labelCustomRadio(name) {
+    return <label htmlFor="c-search_2_refine-campus">{name}</label>
+  }
+
+  render() {
+    let p = this.props,
+        customRadioName = "",
+        customRadioLabel = ""
+    if (["oru", "series", "monograph_series", "seminar_series", "special"].includes(p.type)) {
+       customRadioName = "departments"
+       customRadioLabel = "This department"
+    }
+    if (p.type == 'journal') {
+       customRadioName = "journals"
+       customRadioLabel = "This Journal"
+    }
+    if (p.type == 'campus') {
+       customRadioName = "campuses"
+       customRadioLabel = "This campus"
+    }
+    return (
+    <div className={this.props.refineActive ? "c-search2__refine--active" : "c-search2__refine"}>
+      <input type="radio" id="c-search2__refine-eschol" name={customRadioName} value="" onClick = {this.handleRadioSelect.bind(this)} defaultChecked={true}/>
```

Comment:
Simpler to just do: `onClick = { () => this.props.handleRadioSelect() }` ... saves a bunch of boilerplate.